science/gcal/files/patch-trap-remove-may-month-check-fix.diff

```diff
@@ -0,0 +1,43 @@
+--- src/utils.c.orig	2020-11-27 23:18:52.000000000 +1300
++++ src/utils.c	2020-12-02 20:24:14.000000000 +1300
+@@ -1336,36 +1336,11 @@
+     N_("October"), N_("November"), N_("December")
+   };
+ 
++  if (month == 5) {
+ 
+-  if (month == 5)
+-    {
+-      static char *mayname;
+-      static Bool is_initialized = FALSE;
+-
+-
+-      /*
+-         Now let's remove the trailing BLANK character
+-         from the complete text for the month MAY.
+-       */
+-      if (!is_initialized)
+-	{
+-	  register int len;
+-	  auto char *s = _(name[month]);
+-
+-
+-	  len = (int) strlen (s) - 1;
+-	  mayname = (char *) my_malloc (len,
+-					ERR_NO_MEMORY_AVAILABLE,
+-					__FILE__, ((long) __LINE__) - 2L,
+-					"mayname", 0);
+-	  strcpy (mayname, s);
+-	  while (*(mayname + len) == ' ')
+-	    *(mayname + len--) = '\0';
+-	  is_initialized = TRUE;
+-	}
+-
+-      return (mayname);
+-    }
++      /*** Translators, please translate this as the complete month name "May". */
++      return N_("May");
```

Comment:
This won't work either. You can't just add a comment that says "translate this string as complete month name". Simply put, translators produce a table in the format "Original string" -> "Localized string", so generally you can't translate the same "Original string" to multiple "Localized string"'s in the same file. That's why that extra space character is needed - to form separate "Original string"'s for full and abbreviated month names. This is not to mention the fact that no one will redo all existing translations for all supported languages to add this particular string here. Such broad changes should be avoided.

Reply:
Wow! Thanks again for explaining. I am going to take a look at how localization works.

In the meantime, if Giuseppe creates a release with your fix, I can update the patch here. Or perhaps we can close this and you can create a pull request.

What suits you?

Reply:
I'm really not very interested in maintaining gcal port, I have my own ports to maintain :) I strongly doubt that Giuseppe will create a separate release containing only my fix, however, you can include this fix in the port patch set (instead of commenting out the code that removes trailing spaces from the month name), verify the result, squash commits and force-push the result.

Reply:
Okay. Sounds reasonable. I have updated the port patch to include your patch from upstream. Thanks again.